openbas-api/src/main/java/io/openbas/rest/tag_rule/TagRuleApi.java

```diff
@@ -0,0 +1,91 @@
+package io.openbas.rest.tag_rule;
+
+import static io.openbas.database.model.User.ROLE_ADMIN;
+
+import io.openbas.aop.LogExecutionTime;
+import io.openbas.rest.helper.RestBehavior;
+import io.openbas.rest.tag_rule.form.TagRuleInput;
+import io.openbas.rest.tag_rule.form.TagRuleMapper;
+import io.openbas.rest.tag_rule.form.TagRuleOutput;
+import io.openbas.service.TagRuleService;
+import io.openbas.utils.pagination.SearchPaginationInput;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotBlank;
+import java.util.List;
+import org.springframework.data.domain.Page;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class TagRuleApi extends RestBehavior {
+
+  public static final String TAG_RULE_URI = "/api/tag-rules";
+  private final TagRuleService tagRuleService;
+
+  public TagRuleApi(TagRuleService tagRuleService) {
+    super();
+    this.tagRuleService = tagRuleService;
+  }
+
+  @LogExecutionTime
+  @GetMapping(TagRuleApi.TAG_RULE_URI + "/{tagRuleId}")
+  public TagRuleOutput findTagRule(@PathVariable @NotBlank final String tagRuleId) {
+    return tagRuleService.findById(tagRuleId).map(TagRuleMapper::toTagRuleOutput).orElse(null);
```

Comment:
**suggestion:**
To improve the performance of these queries, it would be better to go directly through the output DTO rather than remap lazy or eager objects after querying them.

Example here:
1. hibernate query
2. map model to DTO with Lazy properties
2. new queries has be done to access LAZY properties
3. Return of the DTO

It would be better to have:
1. custom query or criteria builder
2. direct mapping to the DTO
3. Return of the DTO